MmSupervisorPkg/Test/MmSupvRequestUnitTestApp/MmSupvRequestUnitTestApp.c

```diff
@@ -0,0 +1,821 @@
+*/
+UNIT_TEST_STATUS
+EFIAPI
+RequestSecurityPolicy (
+  IN UNIT_TEST_CONTEXT  Context
+  )
+{
+  SMM_SUPV_SECURE_POLICY_DATA_V1_0  *SecurityPolicy;
+
+  SecurityPolicy = FetchSecurityPolicyFromSupv (Context);
+
+  UT_ASSERT_NOT_NULL (SecurityPolicy);
+
+  return UNIT_TEST_PASSED;
+}
+
+/*
+  Test case to inspect requested policy
+*/
+UNIT_TEST_STATUS
+EFIAPI
+InspectSecurityPolicy (
+  IN UNIT_TEST_CONTEXT  Context
+  )
+{
+  SMM_SUPV_SECURE_POLICY_DATA_V1_0  *SecurityPolicy;
+  SMM_SUPV_POLICY_ROOT_V1           *PolicyRoot;
+  EFI_STATUS                        Status;
+  UINTN                             Index0;
+  UINT32                            MsrLevel;
+  UINT32                            IoLevel;
+  UINT32                            MemLevel;
+  UINT32                            SvstLevel;
+  UINT32                            FinalLevel;
+
+  MsrLevel   = UNDEFINED_LEVEL;
+  IoLevel    = UNDEFINED_LEVEL;
+  MemLevel   = UNDEFINED_LEVEL;
+  SvstLevel  = UNDEFINED_LEVEL;
+  FinalLevel = 0;
+
+  SecurityPolicy = FetchSecurityPolicyFromSupv (Context);
+
+  UT_ASSERT_NOT_NULL (SecurityPolicy);
+
+  UT_ASSERT_EQUAL (SecurityPolicy->VersionMajor, 0x0001);
+
+  PolicyRoot = (SMM_SUPV_POLICY_ROOT_V1 *)((UINTN)SecurityPolicy + SecurityPolicy->PolicyRootOffset);
+  for (Index0 = 0; Index0 < SecurityPolicy->PolicyRootCount; Index0++) {
+    if ((PolicyRoot[Index0].AccessAttr != SMM_SUPV_ACCESS_ATTR_ALLOW) &&
+        (PolicyRoot[Index0].AccessAttr != SMM_SUPV_ACCESS_ATTR_DENY))
+    {
+      continue;
+    }
+
+    switch (PolicyRoot[Index0].Type) {
+      case SMM_SUPV_SECURE_POLICY_DESCRIPTOR_TYPE_IO:
+        if (IoLevel != UNDEFINED_LEVEL) {
+          Status = EFI_ALREADY_STARTED;
+          break;
+        }
+
+        Status = VerifyIoPolicy ((UINT8 *)SecurityPolicy + PolicyRoot[Index0].Offset, PolicyRoot[Index0].Count, PolicyRoot[Index0].AccessAttr, &IoLevel);
+        break;
+      case SMM_SUPV_SECURE_POLICY_DESCRIPTOR_TYPE_MSR:
+        if (MsrLevel != UNDEFINED_LEVEL) {
+          Status = EFI_ALREADY_STARTED;
+          break;
+        }
+
+        Status = VerifyMsrPolicy ((UINT8 *)SecurityPolicy + PolicyRoot[Index0].Offset, PolicyRoot[Index0].Count, PolicyRoot[Index0].AccessAttr, &MsrLevel);
+        break;
+      case SMM_SUPV_SECURE_POLICY_DESCRIPTOR_TYPE_MEM:
+        if (MemLevel != UNDEFINED_LEVEL) {
+          Status = EFI_ALREADY_STARTED;
+          break;
+        }
+
+        Status = VerifyMemPolicy ((UINT8 *)SecurityPolicy + PolicyRoot[Index0].Offset, PolicyRoot[Index0].Count, PolicyRoot[Index0].AccessAttr, &MemLevel);
+        break;
+      case SMM_SUPV_SECURE_POLICY_DESCRIPTOR_TYPE_SAVE_STATE:
+        if (SvstLevel != UNDEFINED_LEVEL) {
+          Status = EFI_ALREADY_STARTED;
+          break;
+        }
+
+        Status = VerifySvstPolicy ((UINT8 *)SecurityPolicy + PolicyRoot[Index0].Offset, PolicyRoot[Index0].Count, PolicyRoot[Index0].AccessAttr, &SvstLevel);
+        break;
+      default:
+        // Do nothing

```

Comment:
in the default case, Status will be uninitialized and the check below will check against the uninitialized value

Reply:
Thanks, will update.

Reply:
Updated